### PR TITLE
ADD: renaming tests

### LINKED ISSUE #341: Add tests for renaming

### PR DESCRIPTION
**Pull request by realemx**
closes #341

### CODE REVIEW

File: src/main/java/mops/businesslogic/directory/DirectoryServiceImpl.java [hunk 1]
```diff
@@ -345,7 +345,7 @@ public long getTotalDirCount() throws MopsException {
     @Override
     @SuppressWarnings({ "PMD.AvoidReassigningParameters", "PMD.LawOfDemeter" })
     public Directory renameDirectory(Account account, long dirId, String newName) throws MopsException {
-        if (newName.isEmpty()) {
+        if (newName.replaceAll("\\s+", "").isEmpty()) {
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/directory/DirectoryServiceImpl.java]: can maybe use String.trim()

File: src/main/java/mops/businesslogic/file/FileServiceImpl.java [hunk 2]
```diff
@@ -268,7 +268,7 @@ public void deleteFile(long fileId) throws MopsException {
     @Override
     @SuppressWarnings({ "PMD.LawOfDemeter", "PMD.AvoidLiteralsInIfCondition", "PMD.AvoidReassigningParameters" })
     public Directory renameFile(Account account, long fileId, String newName) throws MopsException {
-        if (newName.isEmpty()) {
+        if (newName.replaceAll("\\s+", "").isEmpty()) {
```
Comment (iTitus) [inline src/main/java/mops/businesslogic/file/FileServiceImpl.java]: can maybe use String.trim()

File: src/main/java/mops/businesslogic/directory/DirectoryServiceImpl.java (hunk 1 above)
Comment (realemx) [reply src/main/java/mops/businesslogic/directory/DirectoryServiceImpl.java]: String.trim() only removes trailing white spaces. .replaceAll("\\s+", "") removes all non printable characters. .trim() would leave a \t as filename, which I would consider empty.

Comment (iTitus) [reply src/main/java/mops/businesslogic/directory/DirectoryServiceImpl.java]: fair

Comment (realemx) [reply src/main/java/mops/businesslogic/directory/DirectoryServiceImpl.java]: I just realized that a tab char or white space would simply be replaced by an underscore. Maybe we could completely ignore non printable characters as they'll be replaced anyway :thinking:

Comment (iTitus) [reply src/main/java/mops/businesslogic/directory/DirectoryServiceImpl.java]: \s will match any whitespace character ´[ \t\n\x0B\f\r]´, while `.trim()` will remove every leading or trailing ascii char with codepoint <= 0x20. That is every control character and the space character. This includes all non printable characters except for `DEL` at the end of the ascii table.

but newName.isEmpty() should be enough, as every odd character gets replaced either way 🤷‍♀

Comment (realemx) [reply src/main/java/mops/businesslogic/directory/DirectoryServiceImpl.java]: It's gone now completely.

**Pull request merged**